### PR TITLE
fix: small objects prompt failure

### LINKED ISSUE #1125: [Bug]: `RuntimeError: stack expects each tensor to be equal size` when annotating small objects (Visual prompt mode)
### Description

The pipeline crashes during processor rebuild when a visual prompt contains multiple labels and at least one polygon annotation is too small to cover any encoder patch cell (16×16 px) after downsampling.

Root cause: a mask too small to cover any patch cell produced a `[0, 384]` empty tensor instead of `[1, 384]`, breaking `torch.stack` when mixed with valid categories.

### Steps to Reproduce

1. Use a portrait-oriented video/image as the source and Matcher/SoftMatcher/PerDINO as the active model.
2. Create two labels, e.g. `cat` and `led`.
3. Draw a **large polygon** for `cat` and a **tiny polygon** (few pixels) for `led`.
4. Create the prompt.

```
RuntimeError: stack expects each tensor to be equal size, but got [1, 384] at entry 0 and [0, 384] at entry 1
  masked_feature_extractor.py, line 171, in forward
    masked_ref_embeddings=torch.stack(masked_ref_embeddings_list, dim=0)
```

### Component

Library (Prompt framework)

### PR DESCRIPTION
**Pull request by LenaRudaya**
Fixes a crash in the prompt/feature-extraction pipeline when a visual prompt includes a category whose polygon mask becomes empty after downsampling to the encoder patch grid (Issue #1125). The change ensures per-category masked reference embeddings remain stackable even when a category has zero patch coverage.

**Changes:**
- Return a zero-filled `[1, embed_dim]` masked reference embedding for zero-coverage masks so `torch.stack` across categories does not fail.
- Add/adjust unit tests to guard against the multi-category “empty masked embedding” regression.

## Type of Change

- [ ] ✨ `feat` - New feature
- [x] 🐞 `fix` - Bug fix
- [ ] 📚 `docs` - Documentation
- [ ] ♻️ `refactor` - Code refactoring
- [ ] 🧪 `test` - Tests
- [ ] 🔧 `chore` - Maintenance

## Related Issues

Fixes #1125

### CODE REVIEW

**Review by michalad1 — DISMISSED** [review]
LGTM!

File: library/src/instantlearn/components/feature_extractors/masked_feature_extractor.py [hunk 1]
```diff
@@ -135,10 +138,23 @@ def forward(
...
+                logger.warning(
+                    "Category id=%d: annotation mask covers zero encoder patch cells "
+                    "after downsampling to the %dx%d patch grid. "
+                    "The polygon may be too small relative to the image size. "
+                    "A zero embedding will be used; this category will not be detected.",
+                    cat_id,
+                    int(self.input_size // self.patch_size),
+                    int(self.input_size // self.patch_size),
+                )
                 averaged_embed = torch.zeros(
```
Comment (rajeshgangireddy) [inline library/src/instantlearn/components/feature_extractors/masked_feature_extractor.py:156]: Hi, can you check if the whole end to end example is also working fine. I suspect,   the predict step might have issue/fail when there are multiple labels and when an annotation is too small? 
Some changes to the SamDecoder at the prediction step might be needed.
